modules/navier_stokes/src/fvkernels/INSFVMixingLengthScalarDiffusion.C

```diff
@@ -0,0 +1,106 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "INSFVMixingLengthScalarDiffusion.h"
+
+registerMooseObject("NavierStokesApp", INSFVMixingLengthScalarDiffusion);
+
+InputParameters
+INSFVMixingLengthScalarDiffusion::validParams()
+{
+  InputParameters params = FVFluxKernel::validParams();
+  params.addClassDescription("Computes the turbulent diffusive flux that appears in "
+                             "Reynolds-averaged fluid conservation equations.");
+  params.addRequiredCoupledVar("u", "The velocity in the x direction.");
+  params.addCoupledVar("v", "The velocity in the y direction.");
+  params.addCoupledVar("w", "The velocity in the z direction.");
+  params.addRequiredCoupledVar("mixing_length", "The turbulent mixing length.");
+  params.addRequiredParam<Real>(
+      "schmidt_number",
+      "The turbulent Schmidt number (or turbulent Prandtl number if the passive scalar is energy) "
+      "that relates the turbulent scalar diffusivity to the turbulent momentum diffusivity.");
+  params.set<unsigned short>("ghost_layers") = 2;
+  return params;
```

Comment:
so for the energy equation you want something like: rho * cp * eddy_diff * grad_T right? 

I dont think rho and cp come in through the Schmidt number but I may be wrong

Reply:
Yeah, `rho` and `cp` shouldn't be in the turbulent Prandtl number. For all the simulations that I'm running, I've divided the energy equation by `rho * cp` so that it doesn't appear in the advection and diffusion terms. In the future I guess we should make versions of this function that multiply by `rho * cp` (if the variable is temperature) or `rho` (if the variable is specific enthalpy)

Reply:
They also could be evaluated on the face, which brings us to #16809

If we have a way ti get gradients of matprop on the face that could fix it